src/client.rs

```diff
@@ -354,49 +354,23 @@ impl Client {
         // Upload layers
         stream::iter(layers)
             .map(|layer| {
+                // This avoids moving `self` which is &mut Self
+                // into the async block. We only want to capture
+                // as &Self
                 let this = &self;
                 async move {
                     let digest = layer.sha256_digest();
-                    match this
-                        .push_blob_chunked(image_ref, &layer.data, &digest)
-                        .await
-                    {
-                        Err(OciDistributionError::SpecViolationError(violation)) => {
-                            warn!(
-                                ?violation,
-                                "Registry is not respecting the OCI Distribution \
-                                       Specification when doing chunked push operations"
-                            );
-                            warn!("Attempting monolithic push");
-                            this.push_blob_monolithically(image_ref, &layer.data, &digest)
-                                .await?;
-                        }
-                        Err(e) => return Err(e),
-                        _ => {}
-                    };
-
-                    Ok(())
+                    this.push_blob(image_ref, &layer.data, &digest).await?;
+                    Result::Ok(())
                 }
             })
-            .boxed() // Workaround to rustc issue https://github.com/rust-lang/rust/issues/104382
```

Comment:
Removal of this workaround breaks my code (#90). Seems that a test case is needed to prevent such changes. But I currently don't know how to write a test case for the issue...

Reply:
ouch, I'm sorry about that. I relied on the unit tests working properly to remove that. I was expecting a compilation error to be raised.

I'll bring the fix back

Reply:
@linyinfeng done with https://github.com/krustlet/oci-distribution/pull/95